apps/portals/e2e/explore.spec.ts

```diff
@@ -108,6 +113,70 @@ const getAvailableFacetsDiv = (page: Page) => {
     .locator('> div')
     .filter({ hasText: 'Available Facets' })
 }
+const toggleChartVisibilityAndVerifyCount = async (page: Page) => {
+  const chartsPanel = page.locator('.FacetNav')
+
+  const allChartCells = chartsPanel.locator('.FacetNavPanel')
+  const allChartsCount = await allChartCells.count()
+
+  const chartCells = chartsPanel.locator('.FacetNavPanel:visible')
+  const initialCount = await chartCells.count()
+  await test.step('confirm chat visibility and count', async () => {
+    let foundChartIndex = -1
+    for (let i = 0; i < allChartsCount; i++) {
+      const chart = allChartCells.nth(i)
+      if (await chart.isVisible()) {
+        foundChartIndex = i
+        break
+      }
+    }
+    const chartCell = allChartCells.nth(foundChartIndex)
+    const chartTitle = chartCell.locator('.FacetNavPanel__title')
+    await chartTitle
+      .getByRole('button', { name: 'Hide graph under Show More' })
+      .click()
+
+    await expect(chartCell).toBeHidden()
+
+    const newCount = await page.locator('.FacetNavPanel:visible').count()
+    expect(newCount).toBe(initialCount - 1)
+  })
+}
+
+const toggleShowHideVisualizationsAndVerify = async (
+  page: Page,
+  shouldBeVisible: boolean,
+) => {
+  const showHideButton = page.getByRole('button', {
+    name: 'Show / Hide Visualizations',
+  })
+  const ageDistChart = page.locator('text=Age Distribution')
+  const isAgeDistributionChartPresent = (await ageDistChart.count()) > 0
+
+  const chartsPanel = page.locator('.FacetNav')
+  const chartCells = chartsPanel.locator('.FacetNavPanel:visible')
+  const initialCount = await chartCells.count()
+
+  await test.step('toggle visualization and count', async () => {
+    for (let i = 0; i < 2; i++) {
```

Comment:
Why is this for loop here? Should this be removed, or am I missing where `i` is used?

Reply:
I wanted to simulate toggling the Show / Hide Visualization twice to look at scenarios when all the visuals are hidden and when it's toggled back to show all the visuals. I also wanted to check whether or not the Show / Hide Visualization button preserve the number of visuals from before toggling. Would it be better to get rid of the loop and repeat the code twice? or making a helper function to extract duplicate logic?

Reply:
That makes sense. The loop is confusing to me, so I like the idea to add a helper function, that you call twice